core/tests/test_int3.h

```diff
@@ -0,0 +1,175 @@
+// -------------------------------------------------------------
+//  Cubzh Core Unit Tests
+//  test_int3.h
+//  Created by Xavier Legland on October 17, 2022.
+// -------------------------------------------------------------
+
+#pragma once
+
+#include "int3.h"
+
+// functions that are NOT tested:
+// int3_op_substract
+// int3_op_div_int
+
+// test that the first int3 is set to { 0, 0, 0 }
+void test_int3_pool_pop(void) {
+    int3* i3 = int3_pool_pop();
+
+    TEST_CHECK(i3->x == 0);
+    TEST_CHECK(i3->y == 0);
+    TEST_CHECK(i3->z == 0);
+
+    int3_free(i3);
+}
+
+// check that there is no crash
+void test_int3_pool_recycle(void) {
+    int3* a = int3_pool_pop();
+    int3_pool_recycle(a);
+    int3* b = int3_pool_pop();
+    int3* c = int3_pool_pop();
+
+    TEST_CHECK(b != NULL);
+    TEST_CHECK(c != NULL);
+
+    int3_pool_recycle(b);
+    int3_pool_recycle(c);
+}
+
+// check that the values correspond to arguments
+void test_int3_new(void) {
+    int3* i3 = int3_new(4, 5, 6);
+
+    TEST_CHECK(i3->x == 4);
+    TEST_CHECK(i3->y == 5);
+    TEST_CHECK(i3->z == 6);
+
+    int3_free(i3);
+}
+
+// check that the values are independent
+void test_int3_new_copy(void) {
+    int3* source = int3_pool_pop();
+    source->x = 2;
+    int3* copy = int3_new_copy(source);
+
+    TEST_CHECK(copy->x == 2);
+    source->x = 10;
+    TEST_CHECK(copy->x == 2);
+
+    int3_pool_recycle(source);
+    int3_pool_recycle(copy);
```

Comment:
maybe you should `free` them here, so that this test doesn't change the state of the program.